### PR TITLE
[GCI] Add button for copying section links

### LINKED ISSUE #91: Add a button to each section to copy link to the section
**New Feature**
The Interactive-Book comprises of various sections and adding a button to copy a link will help in sharing a specific section.

Something like this [link](https://developers.google.com/open-source/gci/resources/getting-started#4_easy_steps)

Links can be added to each and every section or to some specific types of content.

**Are you working on this issue?**
Nope. For GCI 2019

### PR DESCRIPTION
**Pull request by junron**
Fixes #91 

## Features
- Same link icon as Google Developers site (material design)
- Icon changes color when hovered
- Icon changes to tick when link is copied
- Icon does not appear if header element has no id
- Shows "Copy link to this section" when button is hovered for some time.

## Demo video
Demo video

Please review @tachyons  @satu0king 

[GCI task](https://codein.withgoogle.com/dashboard/task-instances/6263525673009152/)

### CODE REVIEW

**Comment by junron** [comment]
@sal2701 Please have a look

**Comment by junron** [comment]
@sal2701 Please review

**Comment by junron** [comment]
@tachyons @satu0king @sal2701 Please have a look.

File: assets/js/section-links.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const buildUrl = id => {
+  const urlParts = location.href.split("#");
+  if (urlParts.length > 1) urlParts.pop();
+  return urlParts.concat("#" + id).join("");
+};
+
+// Source:
+// https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f
+const copyToClipboard = str => {
```
Comment (tachyons) [inline assets/js/section-links.js]: Browser compatibility can be an issue here since this es6 syntax

Comment (junron) [reply assets/js/section-links.js]: @tachyons Resolved!

**Comment by junron** [comment]
@tachyons Can the task be approved?

**Comment by tachyons** [comment]
Looks good to me @sal2701 please test and approve

**Comment by YashKumarVerma** [comment]
There is an issue here, both #152 and #155 provide the same functionality, links to sections.

But, since
#152 
- has single commit
- has less files edited
- was opened before
- changes the cursor to a pointer  

whereas in #155 

- has multiple commits
- has more files edited
- was opened later
- changes the cursor to text type
- included jquery as a library

therefore keeping the conventions and good practices in mind, #152 is being merged and #155 is closed its favor.

Thanking @JoshHeng  and @junron  for their contribution.

**Pull request closed without merging**